src/plugins/score-plugin-avnd/Spat/MaSpat.hpp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+#include <halp/audio.hpp>
+#include <halp/controls.hpp>
+#include <halp/meta.hpp>
+
+#include <iostream>
+
+#include <experimental/mdspan>
+
+class MaSpat
+{
+public:
+  halp_meta(name, "MaSpat") halp_meta(c_name, "avnd_maspat")
+      halp_meta(uuid, "82bdb9c5-9cf8-440e-8675-c0caf4fc59b9")
+
+          using setup = halp::setup;
+  using tick = halp::tick;
+
+  struct
+  {
+    halp::dynamic_audio_bus<"Input", double> audio;
+    halp::hslider_f32<"L/R", halp::range{.min = -1, .max = 1, .init = 0}> toto;
+  } inputs;
+
+  struct
+  {
+    halp::dynamic_audio_bus<"Output", double> audio;
+  } outputs;
+
+  /* MaSpat()
+    {
+
+    }
+
+    ~MaSpat()
+    {
+
+    }*/
+
+  halp::setup setup_info;
+  void prepare(halp::setup info)
+  {
+    previous_values.resize(info.input_channels);
+  }
+
+  void operator()(halp::tick t)
+  {
+    if (inputs.audio.channels == 2 && outputs.audio.channels == 2)
```

Comment:
So, here instead of checking at runtime what you can do is use 

    halp::fixed_audio_bus<"Input", double, 2> audio;

as a data type: this way the engine will make sure that you always get two input channels and you won't have to check for it.